applications/build-modular-monolith-with-platformatic/services/books-service/plugin.js

```diff
@@ -0,0 +1,38 @@
+/// <reference path="./global.d.ts" />
+/// <reference path="./clients/persons-service/persons_service.d.ts" />
+'use strict'
+
+/** @param {import('fastify').FastifyInstance} app */
+module.exports = async function (app) {
+
+  app.platformatic.addEntityHooks('book', {
+    find: async (originalFind, options) => {
+      const books = await originalFind(options)
+
+      const personIds = books.map(({ authorId }) => authorId)
+      const persons = await app.persons_service.getPersons({ "where.id.in": personIds })
+
+      const getPersonNameById = (id) => {
+        const person = persons.find(person => person.id === id)
+        if (person) {
+          return person.name
+        }
+        return null
+      }
+
+      // TODO: The added `authorName` field is stripped out by the serialization schema
+      //       for the Platformatic DB GET routes (/books and /books/:id)
+      //       Is there a way to update the serialization schema for those routes?
+      //       That feels quite clunky, but I can't think of another way.
+      const newBooks = books.map((book) => {
+        book.authorName = getPersonNameById(book.authorId)
```

Comment:
This isn't working. Fastify strips out the `authorName` field as it doesn’t exist in the response schema generated by Platformatic DB for the `GET /books/*` routes.

Is there a way to update the serialization schema for those routes? Or a better approach?

Reply:
How is the find hook not working?

Reply:
I need a way to reproduce the problem if something is not working as expected.

Reply:
@mcollina The `find` hook is firing. The issue is that the field I'm adding in the hook, `authorName`, isn't in the `Book` response schema that's been generated by Platformatic DB. This means it's ignored when the response is serialized by Fastify on the `/books` and `/books/{id}` routes.